torch/csrc/jit/pybind_utils.h

```diff
@@ -287,37 +285,24 @@ inline bool isTraceableType(TypePtr type) {
   return false;
 }
 
-inline TypedIValue toTraceableIValue(py::handle input) {
+inline IValue toTypeInferredIValue(py::handle input) {
   auto match = tryToInferType(input);
   if (!match.success()) {
     AT_ERROR(
         "Tracer cannot infer type of ", py::str(input), "\n:", match.reason());
   }
-  auto type = match.type();
-
-  if (isTraceableType(type)) {
-    return TypedIValue(toIValue(input, type), type);
-  }
+  return toIValue(input, match.type());
+}
 
-  AT_ERROR(
+inline Stack toTraceableStack(const py::tuple& inputs) {
+  auto info = toTypeInferredIValue(inputs);
+  AT_CHECK(
+      isTraceableType(info.type()),
```

Comment:
Hi, @zdevito 
May I know why we only support Tensors and (possibly nested) Lists, Dicts, and Tuples of Tensors?
What if we passed a `None` type of python, and what will happen that makes us not support it?